lib/graphiti/scoping/filter_group_validator.rb

```diff
@@ -0,0 +1,76 @@
+module Graphiti
+  class Scoping::FilterGroupValidator
+    VALID_FILTER_GROUP_REQUIRED_VALUES = %i[all any]
+
+    def initialize(resource, query_hash)
+      @resource = resource
+      @query_hash = query_hash
+    end
+
+    def raise_unless_filter_group_requirements_met!
+      return if grouped_filters.empty?
+
+      case filter_group_requirement
+      when :all
+        raise_unless_all_requirements_met!
+      when :any
+        raise_unless_any_requirements_met!
+      else
+        raise Errors::FilterGroupInvalidRequirement.new(
```

Comment:
If I'm reading this correctly, this is raised if I declare the filter group incorrectly?

```ruby
filter_group [:foo, :bar], no_required_key: :all
```

If so, I think we should do that as part of the DSL method at initialization time instead of at runtime.

Reply:
Yes correct, raises the error if the :required option is invalid. Raising at DSL method initialization sounds good.

Reply:
@wadetandy moved this error check to the DSL method invocation.